src/cattrs/gen.py

```diff
@@ -340,7 +340,7 @@ def make_dict_structure_fn(
             lines.append(f"{i}except Exception as e:")
             i = f"{i}  "
             lines.append(
-                f"{i}e.__note__ = 'Structuring class {cl.__qualname__} @ attribute {an}'"
+                f"{i}e.__note__ = 'Structuring class ' + {cl.__qualname__!r} + ' @ attribute', {an}"
```

Comment:
Technically speaking, the concatenation operators aren't required but it reads better in my opinion. Feel free to change if desired.

Reply:
Turns out I had an error in here- I was originally doing `(' ').join(...)` which I botched when moving to string concatenation instead